front/src/applications/operationalStudies/components/MacroEditor/ngeToOsrd.ts

```diff
@@ -258,155 +265,318 @@ const createTrainSchedulePayload = async ({
+  });
+  const newPacedTrain = await createPacedTrain(
+    timetableId,
+    timetableItemBase as PacedTrainBase,
+    dispatch
+  );
+  state.timetableItemIdByNgeId.set(trainrun.id, newPacedTrain.id);
+  addUpsertedTimetableItems([newPacedTrain]);
+};
+
+const handleUpdateTimetableItem = async ({
   netzgrafikDto,
   trainrun,
   dispatch,
   infraId,
-  addUpsertedTrainSchedules,
+  addUpsertedTimetableItems,
   state,
 }: {
   netzgrafikDto: NetzgrafikDto;
   trainrun: TrainrunDto;
   dispatch: AppDispatch;
   infraId: number;
-  addUpsertedTrainSchedules: (trainSchedules: TrainScheduleResultWithTrainId[]) => void;
+  addUpsertedTimetableItems: (timetableItems: TimetableItemWithTimetableId[]) => void;
   state: MacroEditorState;
-}) => {
-  const { nodes, labels } = netzgrafikDto;
-  const trainrunSectionsByTrainrunId = getTrainrunSectionsByTrainrunId(netzgrafikDto, trainrun.id);
-  // TODO Paced train : Adapt to handle paced trains in issue https://github.com/OpenRailAssociation/osrd/issues/10612
-  const editoastTrainScheduleIdToUpdate = formatTrainScheduleIdToEditoastTrainId(
-    state.trainScheduleIdByNgeId.get(trainrun.id) as TrainScheduleId
-  );
-  const trainSchedule = await dispatch(
-    osrdEditoastApi.endpoints.getTrainScheduleById.initiate({
-      id: editoastTrainScheduleIdToUpdate,
-    })
-  ).unwrap();
-  const startDate = new Date(trainSchedule.start_time);
-  const trainSchedulePayload = await createTrainSchedulePayload({
-    trainrunSections: trainrunSectionsByTrainrunId,
-    nodes,
+}): Promise<void> => {
+  const timetableItemId = state.timetableItemIdByNgeId.get(trainrun.id)!;
+  const timetableItem = await fetchTimetableItem(timetableItemId, dispatch);
+  const timetableItemBase = await generateTimetableItemBase({
+    trainrunSections: getTrainrunSectionsByTrainrunId(netzgrafikDto, trainrun.id),
+    nodes: netzgrafikDto.nodes,
     trainrun,
     infraId,
     dispatch,
-    labels,
-    oldStartDate: startDate,
-    trainSchedule,
+    labels: netzgrafikDto.labels,
+    timetableItem,
   });
-  const newTrainSchedule = await dispatch(
-    osrdEditoastApi.endpoints.putTrainScheduleById.initiate({
-      id: editoastTrainScheduleIdToUpdate,
-      trainScheduleForm: {
-        ...trainSchedule,
-        ...trainSchedulePayload,
-        // Reset margins because they contain references to path items
-        margins: undefined,
-      },
+  const updatedTimetableItem = await updateTimetableItem(
+    timetableItemId,
+    { ...timetableItem, ...timetableItemBase },
+    dispatch
+  );
+  addUpsertedTimetableItems([updatedTimetableItem]);
+};
+
+const handleDeleteTimetableItem = async (
+  trainrunId: number,
+  state: MacroEditorState,
+  dispatch: AppDispatch,
+  addDeletedTimetableItemIds: (timetableItemIds: TimetableItemId[]) => void
+): Promise<void> => {
+  const timetableItemId: TimetableItemId = state.timetableItemIdByNgeId.get(trainrunId)!;
+  const editoastTrainId: number = isPacedTrain(timetableItemId)
```

Comment:
Typescript knows what you are manipulating
```suggestion
  const timetableItemId = state.timetableItemIdByNgeId.get(trainrunId)!;
  const editoastTrainId = isPacedTrain(timetableItemId)
```

Reply:
Done in "remove overtyping"